app/assets/javascripts/admin/addon/controllers/modals/admin-watched-word-test.js

```diff
@@ -16,16 +16,18 @@ export default Controller.extend(ModalFunctionality, {
     "isTag",
     "isLink"
   )
-  matches(value, regexpString, words, isReplace, isTag, isLink) {
-    if (!value || !regexpString) {
+  matches(value, regexpList, words, isReplace, isTag, isLink) {
+    if (!value || !regexpList.length) {
       return [];
     }
 
     if (isReplace || isLink) {
       const matches = [];
       words.forEach((word) => {
-        const regexp = new RegExp(word.regexp, "gi");
+        const caseFlag = word.case_sensitive ? "" : "i";
+        const regexp = new RegExp(word.regexp, `${caseFlag}g`);
```

Comment:
Since we're doing this fairly often (at least 3 times), it might be better to write a small helper that will take a "watched word" as a parameter and return the proper regular expression.